app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -2,22 +2,23 @@
 
   Spree::PermittedAttributes.checkout_attributes << :gift_code
 
-  durably_decorate :update, mode: 'soft', sha: '908d25b70eff597d32ddad7876cd89d3683d6734' do
-    if @order.update_from_params(params, permitted_checkout_attributes)
+  def update
+    if @order.update_from_params(params, permitted_checkout_attributes, request.headers.env)
       if @order.gift_code.present?
         render :edit and return unless apply_gift_code
       end
 
-      persist_user_address
+      @order.temporary_address = !params[:save_user_address]
+
       unless @order.next
         flash[:error] = @order.errors.full_messages.join("\n")
-        redirect_to checkout_state_path(@order.state) and return
+        redirect_to checkout_state_path(@order.state) && return
```

Comment:
@tanmay3011 Its present in spree 3.0.5, haven't found use of it here, removing it.